src/main/java/io/vertx/spi/cluster/hazelcast/HazelcastClusterManager.java

```diff
@@ -506,15 +507,22 @@ public void compareAndSet(long expected, long value, Handler<AsyncResult<Boolean
   private class HazelcastLock implements Lock {
 
     private final ISemaphore semaphore;
+    private final AtomicBoolean released;
 
     private HazelcastLock(ISemaphore semaphore) {
       this.semaphore = semaphore;
+      this.released = new AtomicBoolean(false);
     }
 
     @Override
     public void release() {
       vertx.executeBlocking(future -> {
-        semaphore.release();
+        synchronized (released) {
```

Comment:
Synchronization can be avoided and we should not run a task on the worker pool if released has been called already.

This should do:

```java
      if (released.compareAndSet(false, true)) {
        vertx.executeBlocking(future -> {
          semaphore.release();
          future.complete();
        }, false, null);
      }
```

Reply:
Is there any chance the `executeBlocking` will fail? I was thinking that it should set the flag right after the `release()`.

Reply:
Do you think it still needs a update? I am thinking it’s already good. If you insist, I can do it by the end of this week. (As I am out for several days.)

Reply:
Yes. No worries. I'll go ahead and push the fix.